packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -68,11 +68,22 @@ export function mediaUpload( {
 
 	// Allowed type specified by consumer
 	const isAllowedType = ( fileType ) => {
-		return ( allowedType === '*' ) || startsWith( fileType, `${ allowedType }/` );
+		return ( ! allowedTypes ) ||
+			some(
+				allowedTypes,
+				( allowedType ) => {
+					// If a complete mimetype is specified verify if it matches exactly the mime type of the file.
+					if ( includes( allowedType, '/' ) ) {
+						return allowedType === fileType;
```

Comment:
It looks like this exact match is newish functionality, so would be good to test it in a unit test